### PR TITLE
Update bridge config validation.

### PR DESCRIPTION
**Pull request by vadim-kovalyov**
Context: We need to properly handle EdgeHub faulty bridge configuration.

After some offline discussion, there are two scenarios that we identified:
- First time deployment: if wrong config is set, EdgeHub should reject the twin, report the error via LastDesiredStatus and stop. It should not proceed with startup and open any public ports b/c we are running the risk of losing messages.
- Subsequent deployments: if wrong config is set, EdgeHub should reject the twin, report LastDesiredStatus and continue with previous configuration. This is (I believe) existing behavior.

In this PR I have updated bridge config validation to allow empty settings for `$upstream` bridge and improved error messages.

Also, removed unnecessary `BridgeUpdate::name()` property in favor of `BridgeUpdate::endpoint()`

### CODE REVIEW

File: mqtt/mqtt-bridge/src/config_update.rs [hunk 1]
```diff
@@ -136,7 +136,6 @@ impl BridgeUpdate {
         }
     }
 
-    // TODO update should have name
```
Comment (dmolokanov) [inline mqtt/mqtt-bridge/src/config_update.rs:139]: `name` is not a part of twin config anymore. So I guess we either need to add that to the config, or remove it from rust code.

Comment (vadim-kovalyov) [reply mqtt/mqtt-bridge/src/config_update.rs:139]: Oh, I see. We already have `endpoint()` below. Missed that... I think I'll try to remove `name`.

File: mqtt/mqtt-bridge/src/controller/bridges.rs [hunk 2]
```diff
@@ -58,12 +58,16 @@ impl Bridges {
     }
 
     pub(crate) async fn send_update(&mut self, update: BridgeUpdate) {
-        if let Some(config) = self.config_updaters.get_mut(update.name()) {
-            if let Err(e) = config.send_update(update).await {
-                error!("error sending bridge update {:?}", e);
+        let endpoint = update.endpoint().to_owned();
```
Comment (dmolokanov) [inline mqtt/mqtt-bridge/src/controller/bridges.rs:61]: why `to_owned()`?

Comment (vadim-kovalyov) [reply mqtt/mqtt-bridge/src/controller/bridges.rs:61]: I need to have a copy of an endpoint that I can log in the `error!`, b/c `update` got consumed before that in `send_update()`.

**Review by dmolokanov — APPROVED** [review]
LGTM

**Comment by nyanzebra** [comment]
Did we just do manual testing?

**Comment by vadim-kovalyov** [comment]
> Did we just do manual testing?

Yes, i have tested that manually. Also, I'm adding E2E test here - https://github.com/Azure/iotedge/pull/4814

**Pull request merged**